### PR TITLE
Update metrics-server chart to show in kubectl cluster-info

### PR DESCRIPTION
**Pull request by goshlanguage**
#### What this PR does / why we need it:

This PR adds metrics-server to `kubectl cluster-info`. This seems appropriate, as Heapster used to report in `cluster-info`.

Before:
```
λ ~/ kubectl cluster-info
Kubernetes master is running at https://192.168.0.1:6443
KubeDNS is running at https://192.168.0.1:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
```

After
```
λ ~/ kubectl cluster-info
Kubernetes master is running at https://192.168.0.1:6443
KubeDNS is running at https://192.168.0.1:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
Metrics-server is running at https://192.168.0.1:6443/api/v1/namespaces/kube-system/services/https:metrics-server:/proxy
```

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/metrics-server/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 0.3.1
 description: Metrics Server is a cluster-wide aggregator of resource usage data.
 name: metrics-server
-version: 2.5.1
+version: 2.5.2
```
Comment (olemarkus) [inline stable/metrics-server/Chart.yaml]: Can you bump minor version?

**Review by olemarkus — CHANGES_REQUESTED** [review]
Sorry for missing this one. Just bump the major version and this should be good to go

File: stable/metrics-server/Chart.yaml (hunk 1 above)
Comment (goshlanguage) [reply stable/metrics-server/Chart.yaml]: 👍

Comment (goshlanguage) [reply stable/metrics-server/Chart.yaml]: Actually, why would we update the minor version? Did you mean the patch version? The current version of the chart is already bumped appropriately. If you disagree, can you please elaborate?

Comment (goshlanguage) [reply stable/metrics-server/Chart.yaml]: @olemarkus I'm not seeing any merge conflicts or otherwise and am confused by the above, please advise.

Comment (steven-sheehy) [reply stable/metrics-server/Chart.yaml]: Any new feature should be a minor increase, as @olemarkus suggested. You added a backwards compatible new option `serviceLabels` that didn't exist before and according to [semver](https://semver.org/) that warrants a minor bump.

File: stable/metrics-server/values.yaml [hunk 2]
```diff
@@ -48,8 +48,12 @@ replicas: 1
 
 podAnnotations: {}
 #  scheduler.alpha.kubernetes.io/critical-pod: ''
+#  priorityClassName: system-node-critical
 
-# priorityClassName: system-node-critical
+serviceLabels: {}
+# Enable the next two labels to show metrics-server when running kubectl cluster-info
+#  kubernetes.io/cluster-service: "true"
```
Comment (steven-sheehy) [inline stable/metrics-server/values.yaml]: Any downside to enabling these by default?

Comment (goshlanguage) [reply stable/metrics-server/values.yaml]: I chose not to, as I don't want to prescribe my solution.

Comment (jhoblitt) [reply stable/metrics-server/values.yaml]: Not adding this label by default is probably reasonable as I believe it has no meaning outside of the `kube-system` namespace.  Before I was aware of this pending change, I opened a PR to add the label to the official example deployment: https://github.com/kubernetes-incubator/metrics-server/pull/252

File: stable/metrics-server/templates/metric-server-service.yaml [hunk 3]
```diff
@@ -7,6 +7,9 @@ metadata:
     chart: {{ template "metrics-server.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- with .Values.serviceLabels }}
```
Comment (jhoblitt) [inline stable/metrics-server/templates/metric-server-service.yaml]: As `.Values.service.annotations` has been merged, this should probably be changed to `.Values.service.labels` for consistency.

**Comment by jhoblitt** [comment]
It looks like this PR needs to be rebased, the version string updated, and the label key names changed to be consistent with the current state of the chart.

File: stable/metrics-server/values.yaml [hunk 4]
```diff
@@ -48,8 +48,12 @@ replicas: 1
...
+#  kubernetes.io/name: Metrics-server
```
Comment (tariq1890) [inline stable/metrics-server/values.yaml]:
```suggestion
#  kubernetes.io/name: metrics-server
```

File: stable/metrics-server/README.md [hunk 5]
```diff
@@ -24,4 +24,5 @@ Parameter | Description | Default
 `extraVolumeMounts` | Ability to provide volume mounts to the pod | `[]`
 `extraVolumes` | Ability to provide volumes to the pod | `[]`
 `podAnnotations` | annotations to be added to pods | `{}`
+`serviceLabels` | labels to be added to the metrics-service service | `{}`
```
Comment (tariq1890) [inline stable/metrics-server/README.md]:
```suggestion
`serviceLabels` | labels to be added to the metrics-server service | `{}`
```

**Comment by tariq1890** [comment]
@ryanhartje Can you resolve the merge conflicts and squash the commit history?

File: stable/metrics-server/README.md [hunk 6]
```diff
@@ -29,6 +29,7 @@ Parameter | Description | Default
 `priorityClassName` | Pod priority class | `""`
 `readinessProbe` | Container readiness probe | See values.yaml
 `service.annotations` | Annotations to add to the service | `{}`
+`serviceLabels` | labels to be added to the metrics-server service | `{}`
```
Comment (tariq1890) [inline stable/metrics-server/README.md]:
```suggestion
`service.labels` | labels to be added to the metrics-server service | `{}`
```

**Comment by tariq1890** [comment]
@ryanhartje The PR seems to be failing since you had committed my suggestions effectively making me a co-author. Could you amend the commit to include my signoff as well if you wish to include me as a co-author? I'd leave that to you. (perfectly okay not being a co-author for this as well)

Also, please don't forget to squash the commit history :).

**Comment by ryanhartje** [comment]
whoops, epic fail in my attempts to squash the commit history :eyes:

**Comment by ryanhartje** [comment]
Thanks @tariq1890, @jhoblitt, @steven-sheehy, @olemarkus for your feedback.

File: stable/metrics-server/README.md [hunk 7]
```diff
@@ -29,6 +29,7 @@ Parameter | Description | Default
 `priorityClassName` | Pod priority class | `""`
 `readinessProbe` | Container readiness probe | See values.yaml
 `service.annotations` | Annotations to add to the service | `{}`
+`service.labels` | labels to be added to the metrics-server service | `{}`
```
Comment (tariq1890) [inline stable/metrics-server/README.md]:
```suggestion
`service.labels` | Labels to be added to the metrics-server service | `{}`
```

Comment (tariq1890) [reply stable/metrics-server/README.md]: I recommend applying this manually and force pushing this :).

File: stable/metrics-server/templates/metric-server-service.yaml [hunk 8]
```diff
@@ -7,6 +7,9 @@ metadata:
     chart: {{ template "metrics-server.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- with .Values.service.labels }}
```
Comment (tariq1890) [inline stable/metrics-server/templates/metric-server-service.yaml]: Can we format this properly here?

**Comment by tariq1890** [comment]
Last round of review comments. Your patience is greatly appreciated @ryanhartje!

File: stable/metrics-server/README.md (hunk 7 above)
Comment (goshlanguage) [reply stable/metrics-server/README.md]: why edit this only, as opposed to all of these?

File: stable/metrics-server/templates/metric-server-service.yaml (hunk 8 above)
Comment (goshlanguage) [reply stable/metrics-server/templates/metric-server-service.yaml]: I've updated them all, I think this is what we want here.

**Comment by ryanhartje** [comment]
@tariq1890 no worries, thanks for reviewing

File: stable/metrics-server/README.md (hunk 7 above)
Comment (tariq1890) [reply stable/metrics-server/README.md]: I think you misunderstood my review comment. The uppercasing of the first letter was only meant for the Description, not the value field.

File: stable/metrics-server/README.md [hunk 9]
```diff
@@ -28,9 +28,10 @@ Parameter | Description | Default
 `podAnnotations` | Annotations to be added to pods | `{}`
 `priorityClassName` | Pod priority class | `""`
 `readinessProbe` | Container readiness probe | See values.yaml
-`service.annotations` | Annotations to add to the service | `{}`
-`service.port` | Service port to expose | `443`
-`service.type` | Type of service to create | `ClusterIP`
+`service.Annotations` | Annotations to add to the service | `{}`
```
Comment (tariq1890) [inline stable/metrics-server/README.md]: Can we change all of these service var names back to camelCase please?

Comment (ryanhartje) [reply stable/metrics-server/README.md]: I just missed the port one in the test, let's try this first.

File: stable/metrics-server/README.md (hunk 7 above)
Comment (goshlanguage) [reply stable/metrics-server/README.md]: Yep, I sure did.

**Pull request merged**